compiler/p/codegen/OMRCodeGenerator.cpp

```diff
@@ -3594,26 +3602,25 @@ TR::Register *addConstantToLong(TR::Node * node, TR::Register *srcHighReg, TR::R
 
 TR::Register *addConstantToInteger(TR::Node * node, TR::Register *trgReg, TR::Register *srcReg, int32_t value, TR::CodeGenerator *cg)
    {
-   intParts localVal(value);
-
-   if (localVal.getValue() >= LOWER_IMMED && localVal.getValue() <= UPPER_IMMED)
+   if (value >= LOWER_IMMED && value <= UPPER_IMMED)
       {
-      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi2, node, trgReg, srcReg, localVal.getValue());
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi2, node, trgReg, srcReg, value);
       }
    else
       {
-      int32_t upperLit = localVal.getHighBitsSigned();
-      int32_t lowerLit = localVal.getLowBitsSigned();
-      if (lowerLit < 0)
+      if (0x00008000 == HI_VALUE(value))
          {
-         upperLit++;
-         if (upperLit == 0x8000)
-            upperLit = -upperLit;
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, trgReg, srcReg, 0x7FFF);
+         generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addis, node, trgReg, trgReg, 0x1);
```

Comment:
I think the extra code for handling this case is redundant here. The result of `addConstantToInteger` is treated as an integer, so its upper word is undefined anyway. My previous remark was referring to when that result was being used as an address, so the upper bits being incorrect would be a problem. Truncating the value should be fine here.

Reply:
Okay, I'll change this to truncation.

Reply:
Are we sure it is always treated as integer?

Reply:
@aviansie-ben please see above. I encountered this code again and just wanted to make sure.

Reply:
This value should always be treated as an integer, since this function is meant specifically to be a helper for adding an immediate value to a 32-bit integer in a register. Using this expecting a 64-bit result is a violation of the API for this function. For 64-bit results, `addConstantToLong` is meant to be used instead.

However, you do make a good point that there could be misuses of this function. I went through all uses of this function. It looks like everything's fine in OMR, but I did find a couple misuses of this function in OpenJ9 that seem to be holdovers from the 32-bit days, though these were already broken before this PR was merged. I've opened eclipse/openj9#9884 to track this.

Reply:
@aviansie-ben thanks for checking!